libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs

```diff
@@ -392,9 +396,61 @@ private EvaluatorLookup CustomizedEvaluatorLookup(EvaluatorLookup baseLookup)
                 return new ExpressionEvaluator(name, BuiltInFunctions.Apply(this.TemplateEvaluator(name)), ReturnType.Object, this.ValidTemplateReference);
             }
 
+            const string lgTempalte = "lgTemplate";
+
+            if (name.Equals(lgTempalte))
+            {
+                return new ExpressionEvaluator(lgTempalte, BuiltInFunctions.Apply(this.LgTemplate()), ReturnType.Object, this.ValidateLgTemplate);
+            }
+
             return baseLookup(name);
         };
 
+        // Evaluator for lgTemplate(templateName, ...args) 
+        // normal case we can just use templateName(...args), but lgTemplate is particularly useful when the template name is not pre-known
+        private Func<IReadOnlyList<object>, object> LgTemplate()
+        => (IReadOnlyList<object> args) =>
+        {
+            var templateName = args[0].ToString();
```

Comment:
Shouldn't this be an expression to evaluate or had that already been done?

Reply:
Builtin.Apply already did that

Reply:
It's you Builtin.Apply did the work :)